### PR TITLE
fix panic when doing Ctrl-C in the middle of a attack

### LINKED ISSUE #23: Panic when stopping infinite attack
```
ali/gui/keybinds.go:36
```

### PR DESCRIPTION
**Pull request by brenol**
this should fix #23.

As the Attack method is now sending messages to the 'metricsCh', and it looks at the Context being complete (completely stopping the attack), I simply removed the channel send that was being done to metricsCh in gui/keybinds.go . This fixed the panic and the Ctrl-C issue for me.

### CODE REVIEW

File: attacker/attacker.go [hunk 1]
```diff
@@ -95,7 +95,4 @@ func Attack(ctx context.Context, target string, resCh chan *Result, metricsCh ch
 		}
 	}
 	metrics.Close()
```
Comment (nakabonne) [inline attacker/attacker.go:97]: To show metrics that cannot be measured until the attack is completed, we'd better send after closing metrics.
```suggestion
	metrics.Close()
	metricsCh <- newMetrics(&metrics)
```

**Comment by nakabonne** [comment]
The root problem can be solved by removing the `close(d.metricsCh)` at `gui/drawer.go:106`, so could you get rid of that? (unit tests will fail, but not a problem for now)

**Comment by nakabonne** [comment]
@brenol Thanks for your contribution! It's a great help with multiple issues. Let me just make one thing clear. Your commit and your Github account don't seem to be tied to each other:
image

I'd like to welcome you as a contributor, but as it stands, you don't count as a contributor. Would it be possible to authenticate the Github account on your local PC?

**Comment by brenol** [comment]
> @brenol Thanks for your contribution! It's a great help with multiple issues. Let me just make one thing clear. Your commit and your Github account don't seem to be tied to each other:
> image
> 
> I'd like to welcome you as a contributor, but as it stands, you don't count as a contributor. Would it be possible to authenticate the Github account on your local PC?

Sure, I'll. I knew removing the close() would fix the issue (and it was the first thing I did) but I did it right when I started navigating through the codebase, so I wasn't sure if it would be actually correct.

I'll get that github thingie fixed asap too.

File: attacker/attacker.go (hunk 1 above)
Comment (brenol) [reply attacker/attacker.go:97]: That actually makes a lot of sense. Will get to work on it

**Comment by brenol** [comment]
Flaky test is hard. Thought I could fix it as-is but I wasn't unable to >sadface< .

I have pushed what I believe is a middleground, but it just feels _wrong_.

If you don't have any problem with the failing test, I'll undo those changes.

File: attacker/attacker.go [hunk 2]
```diff
@@ -88,6 +88,8 @@ func Attack(ctx context.Context, target string, resCh chan *Result, metricsCh ch
 		select {
 		case <-ctx.Done():
 			opts.Attacker.Stop()
+			// metricsCh is already closed (as context is done) so we shouldn't send any metric
+			return
```
Comment (nakabonne) [inline attacker/attacker.go:92]: Nice idea! no problem to give back the function when the context expired.

File: gui/drawer.go [hunk 3]
```diff
@@ -103,6 +103,7 @@ End: %v`
 )
 
 func (d *drawer) redrawMetrics(ctx context.Context) {
+	defer close(d.metricsCh)
```
Comment (nakabonne) [inline gui/drawer.go]: To be honest, we want to get rid of it. But let us leave it as is for once.

**Comment by nakabonne** [comment]
@brenol Thank you for taking the time to solve the problem! LTGM so let me merge this once Github account stuff solved.

**Comment by brenol** [comment]
Still not solved? Thought I was able to solve it (set .gitconfig e-mail to my no-reply e-mail; seems I'm still missing something)

**Comment by brenol** [comment]
Now it's correct, right? Sorry for the force push, easiest way to fix it :)

**Review by nakabonne — APPROVED** [review]
Way to go! helped a lot!

**Pull request merged**